torchtext/experimental/vectors.py

```diff
@@ -0,0 +1,188 @@
+import csv
+import logging
+import os
+
+import torch
+from torch import Tensor
+import torch.nn as nn
+from tqdm import tqdm
+
+from torchtext.utils import (
+    download_from_url,
+    extract_archive
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _load_token_and_vectors_from_file(file_path):
+    stoi, tokens, vectors, dup_tokens = {}, [], [], []
+    with open(file_path, 'rb') as f:
+        for line in tqdm(f):
+            token, entries = line.rstrip().split(b" ", 1)
+            vector = torch.tensor([float(c) for c in entries.split(b" ")], dtype=torch.float)
+
+            try:
+                if isinstance(token, bytes):
+                    token = token.decode('utf-8')
+            except UnicodeDecodeError:
+                logger.info("Skipping non-UTF8 token {}".format(repr(token)))
+                continue
+
+            if token in stoi:
+                logger.info("Found duplicate entries for token {} on line {}.".format(repr(token), len(vectors) + 1))
+                dup_tokens.append(token)
+                continue
+
+            stoi[token] = len(vectors)
+            tokens.append(token)
+            vectors.append(vector)
+
+    return tokens, vectors, dup_tokens
+
+
+def fast_text(language="en", unk_tensor=None, root='.data'):
+    r"""Create a fast text Vectors object.
+
+    Args:
+        language (str): the language to use for FastText.
+        unk_tensor (Tensor): a 1d tensor representing the vector associated with an unknown token
+        root: folder used to store downloaded files in (.data)
+
+    Returns:
+        Vectors: a Vectors object.
+
+    Raises:
+        ValueError: if duplicate tokens are found in fasttext file.
+
+    """
+    url = 'https://dl.fbaipublicfiles.com/fasttext/vectors-wiki/wiki.{}.vec'.format(language)
+    vectors_file_path = os.path.join(root, os.path.basename(url) + '.pt')
+    if os.path.isfile(vectors_file_path):
+        return(torch.load(vectors_file_path))
+
+    downloaded_file_path = download_from_url(url, root=root)
+    tokens, vectors, dup_tokens = _load_token_and_vectors_from_file(downloaded_file_path)
+
+    if dup_tokens:
+        raise ValueError("Found duplicate tokens in file: {}".format(str(dup_tokens)))
+
+    vectors_obj = Vectors(tokens, vectors, unk_tensor=unk_tensor)
+    torch.save(vectors_obj, vectors_file_path)
+    return vectors_obj
+
+
+def glo_ve(name="840B", dim=300, unk_tensor=None, root='.data'):
```

Comment:
We should be explicit for the dim options supported. And please include it in the doc.

Reply:
I don't think the `dim` param value is being used for anything actually. In the past it was being used for the cached `.pt` file_name. I've gone ahead and removed it. Lmk if you think its necessary.